pkg/engine/executor/filter.go

```diff
@@ -128,6 +153,11 @@ func filterBatch(batch arrow.Record, include func(int) bool) arrow.Record {
 			builder := array.NewTimestampBuilder(mem, &arrow.TimestampType{Unit: arrow.Nanosecond, TimeZone: "UTC"})
 			builders[i] = builder
 			additions[i] = func(offset int) {
+				if batch.Column(i).IsNull(offset) {
+					builder.AppendNull()
+					return
+				}
```

Comment:
It might be easier to do this in the centralized check below? 

```go 
var ct int64 

for i := 0; i < int(batch.NumRows()); i++ {
  if !include(i) {
    continue 
  }

  for col, add := range additions {
    if batch.Column(col).IsNull(i) {
      builders[col].AppendNull()
      continue 
    }

    add(i) 
  } 
}
```

Reply:
thanks, much better. updating.

Reply:
added in https://github.com/grafana/loki/pull/19214/commits/41df3ddcf5cb2929881683c4d2a04e53f1d9e930